src/lib/jsonParseV2/parseBatchUpdate.cpp

```diff
@@ -104,7 +104,15 @@ std::string parseBatchUpdate(ConnectionInfo* ciP, BatchUpdate* burP)
       // param 4 for parseEntityVector(): attributes are allowed in payload
       std::string r = parseEntityVector(ciP, iter, &burP->entities, true);
 
-      if (r != "OK")
+      if (r == "Unsupported idPattern")
+      {
+        alarmMgr.badInput(clientIp, r);
+        oe.fill(SccNotImplemented, r, "NotImplemented");
+        ciP->httpStatusCode = SccNotImplemented;
+        r = oe.toJson();
+        return r;
+      }
+      else if (r != "OK")
```

Comment:
If you are already detecting the attempt to use idPattern with POST /v2/op/update at the parsing stage (taking into account your modifications in src/lib/jsonParseV2/parseEntityObject.cpp) probably the modification in this file is not needed.

Reply:
As per existing design in `src/lib/jsonParseV2/parseEntityObject.cpp` class, errors are only detected but thrown to other class and the handling of errors have been in other classes. I've also followed the same design for modification in parseEntityObject.cpp class.

There is one more point as if the error handling is removed from  `src/lib/jsonParseV2/parseBatchUpdate.cpp` then expected response is not returning i.e. it returning the below error code 400 whereas the expected code is 501.
```
400 Bad Request  
{"error":"BadRequest","description":"Unsupported idPattern"}
```

Reply:
Ok. So at `src/lib/jsonParseV2/parseEntityObject.cpp` you return `Unsupported idPattern` and that is used here in `src/lib/jsonParseV2/parseBatchUpdate.cpp` to compose the final response to the client. I didn't get it in my first read ;)

Thus NTC